pkg/policies/controlplane/policy-factory.go

```diff
@@ -118,6 +130,7 @@ func (factory *policyFactory) provideControllerPolicyFxOptions(
 
 	// save policy wrapper proto in status registry
 	reg.Child("policy_config").SetStatus(status.NewStatus(&wrapperMessage, nil))
+	factory.policyWrapperList = append(factory.policyWrapperList, &wrapperMessage)
```

Comment:
where are we removing policies when they are removed from the system?

Reply:
let's make this a map that maps policy name to config and we remove this value when the policy is removed.

Reply:
Factory seems to be only spawning policies and it does not track removing them. The only place I found when I could have attached the removing is in the etcd notifier for Remove action. But that would be very ugly solution.
I think I'll close this PR because it's not a priority.

Reply:
You  need to add a tracker for policies. it's simply a map.

Reply:
I've added a map in policy factory instead of list, that's easy, but what tracker are you talking about - new etcd notifier inside policy factory reacting to Remove? wouldn't that be ugly?

Reply:
just make a normal map in policy factory and add remove entries from it when policies load/unload.

Reply:
I will try to explain myself on slack.